auction-server/src/opportunity/entities/opportunity_svm.rs

```diff
@@ -61,15 +61,67 @@ impl PartialEq<ProgramFeeToken> for FeeToken {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum TokenAccountInitializationConfig {
+    Initialized,
+    SearcherPayer,
+    UserPayer,
+}
+
+impl TokenAccountInitializationConfig {
+    pub fn from_optional_balance(balance: Option<u64>, user_payer: bool) -> Self {
+        match balance {
+            Some(_) => Self::Initialized,
+            None => {
+                if user_payer {
+                    Self::UserPayer
+                } else {
+                    Self::SearcherPayer
+                }
+            }
+        }
+    }
+}
+
+
+impl Default for TokenAccountInitializationConfig {
+    fn default() -> Self {
```

Comment:
why is this the default? should there even be a default, or should we always be explicit about the variant we use?

Reply:
you're totally right. I made this for tests but it shouldn't be a `Default`